bootstrap/051_materialization_status.sql

```diff
@@ -0,0 +1,102 @@
+        ('REPORTING', 'WAREHOUSE_DAILY_UTILIZATION', 'QUERY_HISTORY_MAINTENANCE', 'QUERY_HISTORY', 60),
+        ('REPORTING', 'WAREHOUSE_HOURLY_UTILIZATION', 'QUERY_HISTORY_MAINTENANCE', 'QUERY_HISTORY', 60),
+
+        ('REPORTING', 'HYBRID_TABLE_USAGE_HISTORY', 'SIMPLE_DATA_EVENTS_MAINTENANCE', 'HYBRID_TABLE_USAGE_HISTORY', 60),
+        ('REPORTING', 'LOGIN_HISTORY', 'SIMPLE_DATA_EVENTS_MAINTENANCE', 'LOGIN_HISTORY', 60),
+        ('REPORTING', 'MATERIALIZED_VIEW_REFRESH_HISTORY', 'SIMPLE_DATA_EVENTS_MAINTENANCE', 'MATERIALIZED_VIEW_REFRESH_HISTORY', 60),
+        ('REPORTING', 'SERVERLESS_TASK_HISTORY', 'SIMPLE_DATA_EVENTS_MAINTENANCE', 'SERVERLESS_TASK_HISTORY', 60),
+        ('REPORTING', 'SESSIONS', 'SIMPLE_DATA_EVENTS_MAINTENANCE', 'SESSIONS', 60),
+        ('REPORTING', 'TASK_HISTORY', 'SIMPLE_DATA_EVENTS_MAINTENANCE', 'TASK_HISTORY', 60),
+        ('REPORTING', 'WAREHOUSE_METERING_HISTORY', 'SIMPLE_DATA_EVENTS_MAINTENANCE', 'WAREHOUSE_METERING_HISTORY', 60),
+
+        ('REPORTING', 'CLUSTER_SESSIONS', 'WAREHOUSE_EVENTS_MAINTENANCE', 'CLUSTER_SESSIONS', 60),
+        ('REPORTING', 'CLUSTER_SESSIONS_DAILY', 'WAREHOUSE_EVENTS_MAINTENANCE', 'CLUSTER_SESSIONS', 60),
+        ('REPORTING', 'CLUSTER_SESSIONS_HOURLY', 'WAREHOUSE_EVENTS_MAINTENANCE', 'CLUSTER_SESSIONS', 60),
+        ('REPORTING', 'WAREHOUSE_SESSIONS', 'WAREHOUSE_EVENTS_MAINTENANCE', 'WAREHOUSE_SESSIONS', 60),
+        ('REPORTING', 'WAREHOUSE_SESSIONS_DAILY', 'WAREHOUSE_EVENTS_MAINTENANCE', 'WAREHOUSE_SESSIONS', 60),
+        ('REPORTING', 'WAREHOUSE_SESSIONS_HOURLY', 'WAREHOUSE_EVENTS_MAINTENANCE', 'WAREHOUSE_SESSIONS', 60)
+    ) as t(user_schema, user_view, task_name, object_name, task_period_mins)
+    UNION ALL
+    -- Expand all pairs of WAREHOUSE_LOAD_MAINTENANCE + warehouse_name (as object_name)
+    select 'REPORTING', 'WAREHOUSE_LOAD_HISTORY', 'WAREHOUSE_LOAD_MAINTENANCE', object_name, 60 from (select distinct object_name from internal.task_log where task_name = 'WAREHOUSE_LOAD_MAINTENANCE')
+), completed_tasks as (
+    -- The history rows generated every time a task is run and the table that task is managing
+    select task_start, task_finish, success, input, output, task_name, object_name, IFF(INPUT is null, 'FULL', 'INCREMENTAL') as kind, query_id, range_min, range_max,
+    FROM INTERNAL.TASK_LOG
+    WHERE success is not null
+), running_tasks as (
+    select task_start, task_finish, success, input, output, task_name, object_name, IFF(INPUT is null, 'FULL', 'INCREMENTAL') as kind, query_id, range_min, range_max,
+    FROM INTERNAL.TASK_LOG
+    WHERE success is null
+), fullmat as (
+    SELECT ct.task_name, ct.object_name, ct.task_start, ct.task_finish, ct.success,
+        output['SQLERRM'] as error_message, ct.range_min, ct.range_max, ct.query_id,
+    FROM completed_tasks ct
+    WHERE (ct.object_name, ct.task_name, ct.task_start) IN (
+        SELECT object_name, task_name, MAX(task_start)
```

Comment:
this subselect looks odd to me, I don't think I understand the goal well enough to comment on it

Reply:
For each task and full and incremental materialization types, I want to find the row with `max(task_start)` within that task_name and materialization type.

I can add a comment explaining this, but maybe there's a better way?